paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSessionCustomerSheetActivityTest.kt

```diff
@@ -0,0 +1,517 @@
+        ) {
+            savedPaymentMethodsPage.onEditButton().performClick()
+
+            val cbcCard = savedPaymentMethodsPage.onSavedPaymentMethod(last4 = "4242")
+
+            cbcCard.assertIsEnabled()
+            cbcCard.assertHasModifyBadge()
+
+            savedPaymentMethodsPage.onModifyBadgeFor(last4 = "4242").performClick()
+
+            editPage.onRemoveButton().assertDoesNotExist()
+        }
+
+    @Test
+    fun `On detach, should remove payment method & un-shown duplicates from saved PMs screen`() = runTest(
+        cards = listOf(
+            createDuplicateCard(id = "pm_1"),
+            PaymentMethodFactory.card(last4 = "1001"),
+        ),
+        isPaymentMethodRemoveEnabled = true,
+        allowsRemovalOfLastSavedPaymentMethod = true,
+    ) {
+        savedPaymentMethodsPage.onEditButton().performClick()
+        savedPaymentMethodsPage.onRemoveBadgeFor(last4 = "4242").performClick()
+
+        enqueuePaymentMethods(
+            cards = listOf(
+                createDuplicateCard(id = "pm_1"),
+                createDuplicateCard(id = "pm_2"),
+                createDuplicateCard(id = "pm_3"),
+                createDuplicateCard(id = "pm_4"),
+            )
+        )
+
+        enqueueDetachPaymentMethod(id = "pm_1")
+        enqueueDetachPaymentMethod(id = "pm_2")
+        enqueueDetachPaymentMethod(id = "pm_3")
+        enqueueDetachPaymentMethod(id = "pm_4")
+
+        removeDialog.confirm()
+        savedPaymentMethodsPage.waitUntilVisible()
+        savedPaymentMethodsPage.waitForSavedPaymentMethodToBeRemoved(last4 = "4242")
+    }
+
+    @Test
+    fun `On detach from edit screen, should remove payment method & un-shown duplicates from saved PMs screen`() =
+        runTest(
+            cards = listOf(
+                createDuplicateCard(id = "pm_1", addCbcNetworks = true),
+                PaymentMethodFactory.card(last4 = "1001"),
+            ),
+        ) {
+            savedPaymentMethodsPage.onEditButton().performClick()
+            savedPaymentMethodsPage.onModifyBadgeFor(last4 = "4242").performClick()
+
+            enqueuePaymentMethods(
+                cards = listOf(
+                    createDuplicateCard(id = "pm_1", addCbcNetworks = true),
+                    createDuplicateCard(id = "pm_2", addCbcNetworks = true),
+                    createDuplicateCard(id = "pm_3", addCbcNetworks = true),
+                    createDuplicateCard(id = "pm_4", addCbcNetworks = true),
+                )
+            )
+
+            enqueueDetachPaymentMethod(id = "pm_1")
+            enqueueDetachPaymentMethod(id = "pm_2")
+            enqueueDetachPaymentMethod(id = "pm_3")
+            enqueueDetachPaymentMethod(id = "pm_4")
+
+            editPage.onRemoveButton().performClick()
+            removeDialog.confirm()
+
+            savedPaymentMethodsPage.waitUntilVisible()
+            savedPaymentMethodsPage.waitForSavedPaymentMethodToBeRemoved(last4 = "4242")
+        }
```

Comment:
For `CustomerSession`, we always expect duplicates to be removed on the client-side even if we don't load or show them initially. We are supposed to replace this at some point with a dedicated endpoint but so far no backend work has been scheduled to do this. For now, we are handling this on the client-side.